torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -4096,6 +4460,71 @@ def rref_error():
 
         dist.barrier()
 
+    class VanillaPs:
+        def __init__(self, trainers):
+            self.lock = Lock()
+            self.trainers = trainers
+            self.updates = 0
+            self.futures = []
+            self.total = None
+
+        @staticmethod
+        @rpc.functions.async_execution
+        def average(rref, tensor):
+            self = rref.local_value()
+            fut = torch.futures.Future()
+            with self.lock:
+                self.futures.append(fut)
+                if self.total is None:
+                    self.total = tensor
+                else:
+                    self.total += tensor
+                self.updates += 1
+                if self.trainers == self.updates:
+                    for fut in self.futures:
+                        result = self.total / self.trainers
+                        fut.set_result(result)
+            return fut
+
+    def _trainer_func(self, rref, tensor):
+        fut = rref.rpc_async().average(*[rref, tensor])
```

Comment:
any reason this is not just `fut = rref.rpc_async().average(rref, tensor)`?

Reply:
I updated the code.